xrspatial/zonal.py

```diff
@@ -752,14 +600,133 @@ def _crosstab_numpy(
             zone_values = _get_zone_values(values_by_zones, start, end)
             _single_zone_crosstab(
                 zone_values, unique_cats, cat_ids,
-                nodata_values, crosstab_dict, agg
+                nodata_values, crosstab_dict
             )
         start = end
 
+    if TOTAL_COUNT in crosstab_dict:
+        crosstab_dict[TOTAL_COUNT] = np.array(
+            crosstab_dict[TOTAL_COUNT], dtype=np.float32
+        )
+    for j, cat in enumerate(cat_ids):
+        crosstab_dict[cat] = np.array(crosstab_dict[cat])
+
+    # construct output dataframe
+    if agg == 'percentage':
+        # replace 0s with nans to avoid dividing by 0 error
+        crosstab_dict[TOTAL_COUNT][crosstab_dict[TOTAL_COUNT] == 0] = np.nan
+        for cat in cat_ids:
+            crosstab_dict[cat] = crosstab_dict[cat] / crosstab_dict[TOTAL_COUNT] * 100  # noqa
+
     crosstab_df = pd.DataFrame(crosstab_dict)
+    crosstab_df = crosstab_df[['zone'] + [c for c in cat_ids]]
```

Comment:
If `cat_ids` is always a `list` or `ndarray`, can we just replace `[c for c in cat_ids]` with `list(cat_ids)`?